runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6754,6 +6754,23 @@ TR_J9ByteCodeIlGenerator::genNewArray(int32_t typeIndex)
         }
      }
 
+   // Special case for handling Arrays.copyOfRangeByte when the caller of caller is String<init>(AbstractStringBuilder, Void).
+   // The caller is always Arrays.copyOfRange since Arrays.copyOfRangeByte is a private method.
+   if ((comp()->getInlineDepth() > 1) && !comp()->isPeekingMethod() && !generateArraylets
```

Comment:
`comp()->getInlineDepth() > 1` makes sure there are at least 3 methods in the call chain. checking `isPeekingMethod()` and `generateArraylets` may not be necessary since we check call depth and second caller signature.

Reply:
Why `generateArraylets` is not needed? If we are generating arrayLet, we have a different mechanism that deals with array allocation, not sure if we would have support for allocation from non zero TLH.